src/api/app/controllers/webui/request_controller.rb

```diff
@@ -145,7 +145,7 @@ def show
 
   def beta_show
     # TODO: Remove this once request_show_redesign is rolled out
-    raise NotFoundError unless Flipper.enabled?(:request_show_redesign, User.session)
+    raise ActiveRecord::RecordNotFound unless Flipper.enabled?(:request_show_redesign, User.session)
```

Comment:
Why not  just redirecting and show a flash message? This is not about a missing record

Reply:
We followed your suggestion and modified the code. Please have another look.